.github/workflows/deploy-cf-it.yml

```diff
@@ -0,0 +1,135 @@
+      name: Build dist
+      working-directory: deploy-cloud-functions
+      run: |-
+        npm install
+        npm run build
+    - id: deploy
+      uses: ./deploy-cloud-functions
+      with:
+        name: cf-http-gcloud-${{ github.run_number }}
+        runtime: nodejs10
+        entry_point: helloWorld
+        source_dir: ./deploy-cloud-functions/tests/test-node-func/
+    - uses: actions/setup-node@master
+      with:
+        node-version: 12.x
+    - run: npm install
+      working-directory: ./deploy-cloud-functions
+    - name: integration tests
+      run: npm run e2e-tests
+      working-directory: ./deploy-cloud-functions
+      env:
+        URL: ${{ steps.deploy.outputs.url }}
+    - name: integration test clean up
+      run: npm run cleanup
+      working-directory: ./deploy-cloud-functions
+      env:
+        DEPLOY_CF_SA_KEY_JSON: ${{ secrets.DEPLOY_CF_SA_KEY_JSON }}
+        CF_NAME: projects/${{ secrets.DEPLOY_CF_PROJECT_ID }}/locations/us-central1/functions/cf-http-gcloud-${{ github.run_number }}
+
+  b64_json:
+    name: with base64 json creds
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - id: build
+      name: Build dist
+      working-directory: deploy-cloud-functions
+      run: |-
+        npm install
+        npm run build
+    - id: deploy
+      uses: ./deploy-cloud-functions
+      with:
+        name: cf-http-b64-${{ github.run_number }}
+        runtime: nodejs10
+        entry_point: helloWorld
+        source_dir: ./deploy-cloud-functions/tests/test-node-func/
+        credentials: ${{ secrets.DEPLOY_CF_SA_KEY_B64 }}
+    - uses: actions/setup-node@master
+      with:
+        node-version: 12.x
+    - run: npm install
+      working-directory: ./deploy-cloud-functions
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master # Set up ADC to make authenticated request to service
+      with:
+        service_account_email: ${{ secrets.DEPLOY_CF_SA_EMAIL }}
+        service_account_key: ${{ secrets.DEPLOY_CF_SA_KEY_JSON }}
+        export_default_credentials: true
+    - name: integration tests
+      run: npm run e2e-tests
+      working-directory: ./deploy-cloud-functions
+      env:
+        URL: ${{ steps.deploy.outputs.url }}
+    - name: integration test clean up
+      run: npm run cleanup
+      working-directory: ./deploy-cloud-functions
+      env:
+        DEPLOY_CF_SA_KEY_JSON: ${{ secrets.DEPLOY_CF_SA_KEY_JSON }}
+        CF_NAME: projects/${{ secrets.DEPLOY_CF_PROJECT_ID }}/locations/us-central1/functions/cf-http-b64-${{ github.run_number }}
+
+  json:
+    name: with json creds
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - id: build
+      name: Build dist
+      working-directory: deploy-cloud-functions
+      run: |-
+        npm install
+        npm run build
+    - id: deploy
+      uses: ./deploy-cloud-functions
+      with:
+        name: cf-http-json-${{ github.run_number }}
+        runtime: nodejs10
```

Comment:
Future improvement, add more testing for different languages, redeploying, and different triggers

Reply:
@averikitsch I do some cf client testing with the redeploy and different tiggers here. In a subsequent PR I can add a new workflow based test too like in cloud run.
https://github.com/GoogleCloudPlatform/github-actions/blob/2f862bdaf60da63e54e24cd9c7dbac5f9fde9f64/deploy-cloud-functions/tests/cloudFunctionClient.test.ts#L65